### PR TITLE
feat(forge): error hints

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

Some cheatcodes lack context when they fail, but our current setup does not allow us to add very much context without the output of `forge test` becoming super cluttered

## Solution

Allow reverts from our backend and our inspectors to include optional hints

Effectively closes https://github.com/foundry-rs/foundry/issues/928 as some of the more annoying cheatcode reverts are documented in other issues

### CODE REVIEW

**Review by gakonst — COMMENTED** [review]
how do u see `encode_error_with_hints` being integrated?

**Comment by onbjerg** [comment]
@gakonst adding an example rn

**Comment by gakonst** [comment]
OK! This makes sense. Supportive.

File: evm/src/error.rs [hunk 1]
```diff
@@ -28,7 +38,208 @@ pub trait SolError: std::error::Error {
...
+                0x01 => SolidityPanic::Assert,
+                0x11 => SolidityPanic::OverUnderFlow,
+                0x12 => SolidityPanic::DivideByZero,
+                0x21 => SolidityPanic::InvalidEnumCast,
+                0x22 => SolidityPanic::InvalidStorageByteArray,
+                0x31 => SolidityPanic::PopOnEmptyArray,
+                0x32 => SolidityPanic::IndexOutOfBounds,
+                0x41 => SolidityPanic::Alloc,
+                0x51 => SolidityPanic::InvalidPointer,
+                code => SolidityPanic::Unknown(code),
```
Comment (mattsse) [inline evm/src/error.rs:103]: could move this to a separate From<U256> impl

**Review by mattsse — COMMENTED** [review]
this looks pretty, immense improvement

**Comment by gakonst** [comment]
Closing as stale.

**Pull request closed without merging**